rct229/rules/section6/section6rule12.py

```diff
@@ -0,0 +1,102 @@
+
+
+class Section6Rule12(RuleDefinitionListIndexedBase):
+    """Rule 12 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
+
+    def __init__(self):
+        super(Section6Rule12, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, False, True),
+            each_rule=Section6Rule12.BuildingRule(),
+            index_rmr="proposed",
+            id="6-12",
+            description="User building is modeled with daylighting controls directly or through schedule adjustments.",
+            rmr_context="ruleset_model_instances/0/buildings",
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule12.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                each_rule=Section6Rule12.BuildingRule.ZoneRule(),
+                index_rmr="proposed",
+                list_path="$..zones[*]",
+            )
+
+        class ZoneRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section6Rule12.BuildingRule.ZoneRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, False, True),
+                    required_fields={
+                        "$": ["spaces", "surfaces"],
+                        "$.spaces[*]": ["interior_lighting"],
+                        "$.spaces[*].interior_lighting[*]": [
+                            "daylighting_control_type",
+                            "are_schedules_used_for_modeling_daylighting_control",
+                        ],
+                        "$.surfaces[*]": [
+                            "adjacent_to",
+                            "subsurfaces",
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                building_p = context.proposed
+
+                daylight_flag_u = False
+                has_daylight_control_flag = False
+                for subsurface in find_all(
+                    "$..surfaces[*].subsurfaces[?classification != 'DOOR']", building_p
+                ):
+                    daylight_flag_u = True
+
+                interior_lighting_u = False
+                for lighting in find_all("$..spaces[*].interior_lighting", building_p):
+                    interior_lighting_u = lighting[0][
+                        "are_schedules_used_for_modeling_daylighting_control"
+                    ]
+                    if lighting[0]["daylighting_control_type"] != "NONE":
+                        has_daylight_control_flag = True
+
+                return {
+                    "daylight_flag_u": daylight_flag_u,
+                    "has_daylight_control_flag": has_daylight_control_flag,
+                    "interior_lighting_u": interior_lighting_u,
```

Comment:
@weilixu  There is a problem with the RDS. My guess is that they mean to have another flag such as `schedules_used_for_modeling_daylighting_flag` that is set if any space has an interior_lighting with `are_schedules_used_for_modeling_daylighting_control` property set to `True`. This flag would then be used in the Rule Assertion Cases 1 and 2.